sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/BasicWriteJobStatsTrackerMetricSuite.scala

```diff
@@ -44,13 +48,14 @@ class BasicWriteJobStatsTrackerMetricSuite extends SparkFunSuite with LocalSpark
       // but the executionId is indeterminate in maven test,
       // so the `statusStore.execution(executionId)` API is not used.
       assert(statusStore.executionsCount() == 2)
-      val executionData = statusStore.executionsList()(1)
-      val accumulatorIdOpt =
-        executionData.metrics.find(_.name == "number of dynamic part").map(_.accumulatorId)
-      assert(accumulatorIdOpt.isDefined)
-      val numPartsOpt = executionData.metricValues.get(accumulatorIdOpt.get)
-      assert(numPartsOpt.isDefined && numPartsOpt.get == partitions)
-
+      eventually(timeout(10.seconds)) {
```

Comment:
is this change related to this PR?

Reply:
I think it is related due to the fact CI is failed, but it passed in my local.

I this the reason is, the metrics event lightly changed. Before all write file metrics come from event `SparkListenerDriverAccumUpdates`.  Now the metrics can be changed by both task event and `SparkListenerDriverAccumUpdates`.